tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -414,3 +414,147 @@ def test_cmp_full_and_first_diff_mem(network_config,
+
+    # Verify if guest can run commands.
+    exit_code, _, _ = ssh_connection.execute_command("sync")
+    assert exit_code == 0
+
+    logger.info("Create snapshot")
+    # Create a snapshot builder from a microvm.
+    snapshot_builder = SnapshotBuilder(basevm)
+
+    # Create base snapshot.
+    snapshot = snapshot_builder.create([root_disk.local_path()],
+                                       ssh_key,
+                                       SnapshotType.DIFF)
+
+    basevm.kill()
+
+    logger.info("Load snapshot, mem %s", snapshot.mem)
+    # Do not resume, just load, so we can still call APIs that work.
+    microvm, _ = vm_builder.build_from_snapshot(snapshot,
+                                                False,
+                                                True)
+    fail_msg = "The requested operation is not supported after starting " \
+        "the microVM"
+
+    try:
+        microvm.start()
+    except AssertionError as error:
+        assert fail_msg in str(error)
+    else:
+        assert False, "Negative test failed"
+
+    try:
+        microvm.basic_config()
+    except AssertionError as error:
+        assert fail_msg in str(error)
+    else:
+        assert False, "Negative test failed"
+
+    microvm.kill()
+
+
+def test_negative_snapshot_permissions(bin_cloner_path):
+    """Test missing permission error scenarios."""
+    logger = logging.getLogger("snapshot_negative")
+    vm_builder = MicrovmBuilder(bin_cloner_path)
+
+    # Use a predefined vm instance.
+    vm_instance = VMNano.spawn(bin_cloner_path)
+    basevm = vm_instance.vm
+    root_disk = vm_instance.disks[0]
+    ssh_key = vm_instance.ssh_key
+
+    basevm.start()
+
+    logger.info("Create snapshot")
+    # Create a snapshot builder from a microvm.
+    snapshot_builder = SnapshotBuilder(basevm)
+
+    disks = [root_disk.local_path()]
+
+    # Remove write permissions.
+    os.chmod(basevm.jailer.chroot_path(), 0o444)
+
+    try:
+        _ = snapshot_builder.create(disks,
+                                    ssh_key,
+                                    SnapshotType.FULL)
+    except AssertionError as error:
+        # Check if proper error is returned.
+        assert "Permission denied" in str(error)
+    else:
+        assert False, "Negative test failed"
+
+    # Restore proper permissions.
+    os.chmod(basevm.jailer.chroot_path(), 0o744)
+
+    # Create base snapshot.
+    snapshot = snapshot_builder.create(disks,
+                                       ssh_key,
+                                       SnapshotType.FULL)
+
+    logger.info("Load snapshot, mem %s", snapshot.mem)
+
+    basevm.kill()
+
+    # Remove permissions for mem file.
+    os.chmod(snapshot.mem, 0o000)
+
+    try:
+        microvm, _ = vm_builder.build_from_snapshot(snapshot,
+                                                    True,
+                                                    True)
+        microvm.kill()
```

Comment:
this will never be called unless the line above will succeed which is something we do not expect to happen. The effect of microvm.kill() not being called will be that the microvm.logging_thread (which by default is always created) will never be stopped which I actually think is the reason why the tests timeout on arm. You can test this by doing 
```
vm.spawn(log_level='Info', create_logger=False)
```
in `build_from_snapshot`.